### PR TITLE
ia-writer-quattro: init at unstable-2023-06-16

### PR DESCRIPTION
**Pull request by x0ba**
## Description of changes

Added the [iA Writer Quattro](https://github.com/iaolo/iA-Fonts) typeface, as well as adding myself as a maintainer.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by katexochen — CHANGES_REQUESTED** [review]
Hey, thanks for your contribution. Currently,  your package isn't referenced anywhere and thus won't be build. Please add it in `pkgs/top-level/all-packages.nix` or, even better, to `pkgs/by-name`. You can find additional explanation in the Readme of that directory.

Also, please use the package name (`pname`) in the commit message.

File: pkgs/data/fonts/ia-writer-quattro/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ stdenvNoCC
+, fetchFromGitHub
+, lib
+,
+}:
+stdenvNoCC.mkDerivation rec {
+  pname = "ia-writer-quattro";
+  version = "f32c04c3058a75d7ce28919ce70fe8800817491b";
```
Comment (katexochen) [inline pkgs/data/fonts/ia-writer-quattro/default.nix:8]: `version` should be `unstable-YYYY-MM-DD`, with the date of the commit that is used.  See the [pkgs readme](https://github.com/NixOS/nixpkgs/tree/master/pkgs#package-naming) for more information.

You should also use that version in your init commit message: `<pname>: init at <version>`

**Review by katexochen — CHANGES_REQUESTED** [review]

File: pkgs/data/fonts/ia-writer-quattro/default.nix (hunk 1 above)
Comment (x0ba) [reply pkgs/data/fonts/ia-writer-quattro/default.nix:8]: Hi, thanks for the suggestions! I'm relatively new to contributing to open source, so bear with me :D. Should I open a new pull request with your suggestions?

Comment (x0ba) [reply pkgs/data/fonts/ia-writer-quattro/default.nix]: I think I've fixed everything. Please let me know if there's anything else!

File: pkgs/data/fonts/ia-writer-quattro/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+  stdenvNoCC,
+  fetchFromGitHub,
+  lib,
+}:
+stdenvNoCC.mkDerivation rec {
+  pname = "ia-writer-quattro";
+  version = "unstable-2023-06-16";
+  commit = "f32c04c3058a75d7ce28919ce70fe8800817491b";
```
Comment (katexochen) [inline pkgs/data/fonts/ia-writer-quattro/default.nix:9]: `commit` not needed here, should be an attribute of `fetchFromGitHub`.

File: pkgs/data/fonts/ia-writer-quattro/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  stdenvNoCC,
+  fetchFromGitHub,
+  lib,
```
Comment (katexochen) [inline pkgs/data/fonts/ia-writer-quattro/default.nix:4]: `lib` goes fist

File: pkgs/data/fonts/ia-writer-quattro/default.nix
Comment (katexochen) [inline pkgs/data/fonts/ia-writer-quattro/default.nix:1]: This file needs to be formatted:

```sh
nix shell nixpkgs#nixpkgs-fmt
nixpkgs-fmt pkgs/data/fonts/ia-writer-quattro/default.nix
```

**Review by katexochen — CHANGES_REQUESTED** [review]
Package build successfully. :)
Please squash your fixups into the commits they belong to.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -29412,6 +29412,8 @@ with pkgs;
 
   ia-writer-duospace = callPackage ../data/fonts/ia-writer-duospace { };
 
+  ia-writer-quattro = callPackage ../data/fonts/ia-writer-quattro { };
+
```
Comment (Lurkki14) [inline pkgs/top-level/all-packages.nix:29416]: New top-level packages should go to `pkgs/by-name` https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md
```suggestion
```

File: pkgs/data/fonts/ia-writer-quattro/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+stdenvNoCC.mkDerivation rec {
+  pname = "ia-writer-quattro";
+  version = "unstable-2023-06-16";
+  commit = "f32c04c3058a75d7ce28919ce70fe8800817491b";
+
+  src = fetchFromGitHub {
+    owner = "iaolo";
+    rev = commit;
+    repo = "iA-fonts";
+    sha256 = "sha256-2T165nFfCzO65/PIHauJA//S+zug5nUwPcg8NUEydfc=";
```
Comment (Lurkki14) [inline pkgs/data/fonts/ia-writer-quattro/default.nix:15]:
```suggestion
    hash = "sha256-2T165nFfCzO65/PIHauJA//S+zug5nUwPcg8NUEydfc=";
```

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

**Comment by x0ba** [comment]
I'm kinda new to git, and I think i might have messed up the latest commit while trying to squash. is it ok if i close this pull request and open a new one that didnt accidentally commit a bunch of other files?

**Comment by katexochen** [comment]
> I'm kinda new to git, and I think i might have messed up the latest commit while trying to squash. is it ok if i close this pull request and open a new one that didnt accidentally commit a bunch of other files?

Sure, go ahead. :) And feel free to ping me on Matrix @katexochen:matrix.org if you need any help.

**Pull request closed without merging**